presto-tpch/src/test/java/com/facebook/presto/tpch/EstimateAssertion.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpch;
+
+import com.facebook.presto.spi.statistics.Estimate;
+import io.airlift.slice.Slice;
+
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static org.testng.Assert.assertEquals;
+
+class EstimateAssertion
+{
+    private final double tolerance;
+
+    public EstimateAssertion(double tolerance)
+    {
+        this.tolerance = tolerance;
+    }
+
+    public void assertClose(Estimate actual, Estimate expected, String message)
+    {
+        assertClose(toOptional(actual), toOptional(expected), message);
+    }
+
+    private Optional<Double> toOptional(Estimate estimate)
+    {
+        return estimate.isValueUnknown() ? empty() : Optional.of(estimate.getValue());
+    }
+
+    public void assertClose(Optional<?> actual, Optional<?> expected, String message)
+    {
+        assertEquals(actual.isPresent(), expected.isPresent(), message);
+        if (actual.isPresent()) {
+            Object actualValue = actual.get();
+            Object expectedValue = expected.get();
+            assertClose(message, actualValue, expectedValue);
+        }
+    }
+
+    private void assertClose(String message, Object actual, Object expected)
```

Comment:
try to be coherent and please put the same parameters in the same order in all the methods eg. lets put message always as last parameter